### DISCUSSION TITLE
Sharing dependencies between workspaces

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2704] [Q&A] [answered]
Hi all, I was wondering if it's possible in yarn 2 to share dependencies between workspaces.

Use case: I use prettier and a few prettier plugins in every workspace. Right now I need to install them in all my workspaces and it feels repetitive. I do have a workspace just for my prettier config where I could install prettier and the plugins. However, I don't think it's possible with strict dependency scoping to run the prettier binary unless it's listed as an explicit dependency by the workspace running the binary.

Is it possible to export a binary from a workspace so other workspaces can run it? Are there any other ways I can share these dependencies? I would like to just be able to include all the prettier dependencies as a group rather than individually for every project.

Thanks!

**Comment by merceyz** [comment, marked as answer]
You can create a global script and use that in all the workspaces https://yarnpkg.com/getting-started/qa/#how-to-share-scripts-between-workspaces

**Reply by wdfinch** [reply]
Thanks for the reply. Yes I know I can do this. The issue is I can't do `yarn run prettier` unless prettier is listed as a dependency in that workspace. Yarn won't be able to find the bin due to script scoping of packages. So is there a way to get past this?

my script:

`"g:format": "cd $INIT_CWD && yarn run prettier --write '{lib,src}/**/*.{js,ts,tsx}'",`

**Reply by merceyz** [reply]
```diff
-`"g:format": "cd $INIT_CWD && yarn run prettier --write '{lib,src}/**/*.{js,ts,tsx}'",`
+`"g:format": "cd $INIT_CWD && prettier --write '{lib,src}/**/*.{js,ts,tsx}'",`
```

**Reply by wdfinch** [reply]
Thanks!
